pygments/lexers/markup.py

```diff
@@ -619,6 +620,100 @@ def __init__(self, **options):
         self.handlecodeblocks = get_bool_opt(options, 'handlecodeblocks', True)
         RegexLexer.__init__(self, **options)
 
+class OrgLexer(RegexLexer):
+    """
+    For Org Mode markup.
+    """
+    name = 'Org Mode'
+    url = 'https://orgmode.org'
+    aliases = ['org', 'orgmode']
+    filenames = ['*.org']
+    mimetypes = ["text/org"]
+    version_added = '2.2'
+    flags = re.MULTILINE
+    tokens = {
+        'root': [
+            (r'^# .*$', Comment),
+
+            # Headings
+            (r'^(\*)( COMMENT)( .*)$', bygroups(Generic.Heading, Name.Entity, Generic.Strong)),
+            (r'^(\*\*+)( COMMENT)( .*)$', bygroups(Generic.Subheading, Name.Entity, Text)),
+            (r'^(\*)( DONE)( .*)$', bygroups(Generic.Heading, String.Regex, Generic.Strong)),
+            (r'^(\*\*+)( DONE)( .*)$', bygroups(Generic.Subheading, String.Regex, Text)),
+            (r'^(\*)( TODO)( .*)$', bygroups(Generic.Heading, Generic.Error, Generic.Strong)),
+            (r'^(\*\*+)( TODO)( .*)$', bygroups(Generic.Subheading, Generic.Error, Text)),
+            (r'^(\*)( .+?)( :[a-zA-Z0-9_@:]+:)$', bygroups(Generic.Heading, Generic.Strong, Generic.Emph)),
+            (r'^(\*)( .+)$', bygroups(Generic.Heading, Generic.Strong)),
+            (r'^(\*\*+)( .+?)( :[a-zA-Z0-9_@:]+:)$', bygroups(Generic.Subheading, Text, Generic.Emph)),
+            (r'^(\*\*+)( .+)$', bygroups(Generic.Subheading, Text)),
+
+            # Lists
+            (r'^( *)([+-] )(\[[ X]\])( .+)$', bygroups(Text, Keyword, Keyword, using(this))),
+            (r'^( +)(\* )(\[[ X]\])( .+)$', bygroups(Text, Keyword, Keyword, using(this))),
+            # Definition lists
+            (r'^( *)([+-] )([^ \n]+ ::)( .+)$', bygroups(Text, Keyword, Keyword, using(this))),
+            (r'^( +)(\* )([^ \n]+ ::)( .+)$', bygroups(Text, Keyword, Keyword, using(this))),
+            # Unordered lists
+            (r'^( *)([+-] )(.+)$', bygroups(Text, Keyword, using(this))),
+            (r'^( +)(\* )(.+)$', bygroups(Text, Keyword, using(this))),
+            # Ordered lists
+            (r'^( *)([0-9]+[.)])( \[@[0-9]+\])( .+)$', bygroups(Text, Keyword, Generic.Emph, using(this))),
+            (r'^( *)([0-9]+[.)])( .+)$', bygroups(Text, Keyword, using(this))),
+
+            # Dynamic Blocks
+            (r'(?i)^( *#\+begin: )([^ ]+)([\w\W]*?\n)([\w\W]*?)(^ *#\+end: *$)', bygroups(Comment, Comment.Special, Comment, using(this), Comment)),
+
+            # Comment Blocks
+            (r'(?i)^( *#\+begin_comment *\n)([\w\W]*?)(^ *#\+end_comment *$)', bygroups(Comment, Comment, Comment)),
+            # Source Code Blocks
+            (r'(?i)^( *#\+begin_src )([^ \n]+)(.*?\n)([\w\W]*?)(^ *#\+end_src *$)', bygroups(Comment, Comment.Special, Comment, using(this), Comment)),
```

Comment:
This could certainly be improved to use the appropriate lexer to highlight the code inside the code block (like it's done in the Markdown lexer), but this is a bit beyond my expertise. I would welcome contributions.

The usual code block format is like this (lifted from [my literal Emacs config](https://git.sr.ht/~gosha/dotfiles/tree/main/item/.doom.d/config.org#L11)):

```org-mode
#+begin_src emacs-lisp :tangle yes
(setq undo-limit 80000000
      evil-want-fine-undo t
      truncate-string-ellipsis "…")
#+end_src
```